blogs/contractionorder.md

```diff
@@ -0,0 +1,338 @@
+Therefore, if we can find the tree decomposition of the tensor network with minimal treewidth, we can find the optimal contraction order of the tensor network.
+
+To further make use of the theorem, we need to introduce the concepts of line graph, tree width and tree decomposition.
+
+### Line Graph
+
+A formal definition of the [**line graph**](https://en.wikipedia.org/wiki/Line_graph) is as follows:
+
+**Definition 1**. Given a graph G, its line graph $L(G)$ is a graph such that: 
+   1. each vertex of $L(G)$ represents an edge of $G$; 
+   2. two vertices of $L(G)$ are adjacent if and only if their corresponding edges share a common endpoint ("are incident") in $G$. 
+
+That is, it is the intersection graph of the edges of G, representing each edge by the set of its two endpoints.
+
+For a tensor network, we can construct a hypergraph $G$ whose vertices are the tensors and whose hyperedges are the indices. 
+Then the line graph $L(G)$ of the hypergraph $G$ is the graph whose vertices are the indices and whose edges are the tensors, which is a simple graph, as shown in the following figure:
+
+![Fig.1](/assets/treewidth_figs/linegraph.png)
+
+Since we are considering a tensor network, dimension of the indices have to be considered. 
+Therefore, for each vertex of the line graph $L(G)$, we define its weight as $\log_2(d)$, where $d$ is the dimension of the index. 
+
+In this way, any possible tensor can be represented as a subgraph induced by a set of vertices in $L(G)$, and its log2 size is given by the sum of weights of the boundary vertices of the subgraph. For example, the tensor $C$ in the figure above corresponding to the subgraph induced by the vertices $\{b, d, f\}$, and its log2 size is 
+$${dims(b) \times dims(d) \times dims(f)} = 2^{w(b) + w(d) + w(f)}$$
+where $w(*)$ for weights of vertices.
+
+### Tree Decomposition and Tree Width
+
+Intuitively, a [**tree decomposition**](https://en.wikipedia.org/wiki/Tree_decomposition) represents the vertices of a given graph $G$ as subtrees of a tree, in such a way that vertices in $G$ are adjacent only when the corresponding subtrees intersect.
+
+The tree decomposition of a graph is a tree whose nodes are subsets of the vertices of the graph, and the following conditions are satisfied:
+
+1. Each vertex of the graph is in at least one node of the tree.
+2. For each edge of the graph, there is a node of the tree containing both vertices of the edge.
+3. Bags containing the same vertex have to be connected in the tree.
+
+All the nodes of the tree are called **tree bags**, and intersection of two bags is called a **separator**.
+The width of a tree decomposition is the size of the largest bag minus one.
+Clearly, one graph can have multiple tree decompositions, and the **tree width** of a graph is the minimum width of all possible tree decompositions, which is called an optimal tree decomposition.
```

Comment:
This last sentence, I would replace with:

Clearly, one graph can have multiple tree decomposition with different corresponding widths. The **tree width** of a graph is the minimal width of all such decompositions, and a particular decomposition (not necessarily unique) that realises this minimal width is called an optimal tree decomposition.


I assume the "not necessarily unique" addition is correct, though I have not given this much thought.